syntheseus/reaction_prediction/chem/utils.py

```diff
@@ -29,6 +29,19 @@ def remove_atom_mapping(smiles: str) -> str:
     return Chem.MolToSmiles(mol)
 
 
+def remove_stereo_information(mol: Molecule) -> Molecule:
+    return Molecule(Chem.MolToSmiles(mol.rdkit_mol, isomericSmiles=False))
+
+
+def remove_stereo_information_from_reaction(reaction: Reaction) -> Reaction:
+    return Reaction(
+        reactants=Bag([remove_stereo_information(mol) for mol in reaction.reactants]),
+        products=Bag([remove_stereo_information(mol) for mol in reaction.products]),
+        identifier=reaction.identifier,
+        metadata=reaction.metadata,
+    )
+
+
```

Comment:
Will this correctly handle the case when a `SingleProductReaction` is given? It looks like this code will downgrade it to a basic `Reaction` object. Could this cause the reaction comparison logic in `eval_single_step.py` to fail?

Reply:
Yeah, it would downgrade to a `Reaction`. The reason why the eval script still worked is because `remove_stereo_information_from_reaction` was applied to _both_ sides of the comparison so both were getting downgraded. I now changed `remove_stereo_information_from_reaction` so that it retains the type of the input, and added tests to verify this works as expected.

More generally, I find it annoying that a `Reaction` that happens to have a single product is not considered equal to a corresponding `SingleProductReaction`. I think `SingleProductReaction` should be mostly syntax sugar for easier construction and type checking. But if we want to change this (i.e. make the reaction comparison ignore the class and only look at the internals), this would be a separate PR anyway.

Reply:
That's a good point: I agree that changing the comparison between `Reaction` and `SingleProductReaction` objects makes sense, but it would make the most sense as a separate PR. Would there be a natural way to do this that doesn't involve manually writing a comparison for each attribute?

Reply:
Not sure if there's a way to tell a dataclass to do this natively, but I'd just write a simple comparator in the base `Reaction` class that compares the fields; should be simple enough?

Reply:
Sounds good!